### PR TITLE
Fixes #35253 - Make boolean type parameters searchable

### PR DESCRIPTION
**Pull request by girijaasoni**
Searchable value when value is a boolean is being set to "t" or "f" because the searchable_value field is "text" type in the database so when we set a boolean value to a string field in the application (Ruby on Rails), it will be converted to "t" or "f. In this fix, we are formatting the boolean type searchable values to the string before it happens automatically. A few points:

- This fix works for updated or new parameters as the existing ones already have their values saved as "t" or "f"
- If we create a boolean param with name boolparam and value true, we can search with `params.boolparam = true` or `params.boolparam != false` . 
- `params.boolparam = t` will not work.

### CODE REVIEW

**Review by nofaralfasi — APPROVED** [review]
Solves the issue for me.

File: app/models/parameter.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def set_default_key_type
   end
 
   def set_searchable_value
-    self.searchable_value = Parameter.format_value_before_type_cast(value, key_type)
+    self.searchable_value = key_type == 'boolean' ? value.to_s : Parameter.format_value_before_type_cast(value, key_type)
```
Comment (ares) [inline app/models/parameter.rb:86]: Just a question, why is the condition here and not in `Parameter.format_value_before_type_cast`? It seems we handle other types there already.

Comment (girijaasoni) [reply app/models/parameter.rb:86]: I tried that initially but format_value_before_type_cast is used at multiple places. I wasn't sure if it's safe to format boolean type to a string for every place that it's called.

**Comment by girijaasoni** [comment]
As discussed with @ShimShtein , it's advised to have the boolean parameters searchable in both ways (i.e with "t" as well as "true"). However there are some complications with the different kinds of operators and we are working on a fix for it.

**Comment by ShimShtein** [comment]
I'll try to give some context here:

Currently we are storing the value that will be used for search in a field called `searchable_value`. Since it's the same field for all value types, the value type is set to string, and the actual value is serialized there.
In our case, for boolean parameters, the value is serialized as `t` or `f`.
When we search by parameter value, we don't have the value type available to us ([example](https://github.com/theforeman/foreman/blob/a1142ab9ef500e3206be1ff8a390431bc1b32de3/app/models/concerns/hostext/search.rb#L172)). So in the current implementation, the user must specify the serialized value when trying to search for a non-string parameter, hence for booleans, the user has to use `t` and `f`, and other values would not work.
We have two options to deal with the problem at hand:
1. We can do another query while searching and get the value type for the current parameter. Then we will be able to translate the value from the search query to the serialized value in the database. E.g. for search `param.boolparam == true` we will first get the type of `boolparam` (boolean), then translate the search token `true` => `'t'` and then search by the translated value.
2. We can try and infer the parameter type from the value itself: if the token is `true` or `false`, and the operator is `=`, it is probably a boolean parameter, hence we need to translate the value to `'t'` or `'f'`. For all other cases, the search will work the same. The only problem with this solution is that it will bring false positives in case the user searches for the literal string `'true'` or `'false'`.

For now we will try to implement the second option, since it should have less impact on the search times.

File: app/models/parameter.rb (hunk 1 above)
Comment (ares) [reply app/models/parameter.rb:86]: I'd say try to make the list of where it's used and consider the impact for each such item. If we don't find a place where this would cause a problem, we should keep the logic at one place. Otherwise in future, when someones modifies the behavior of `Parameter.format_value_before_type_cast` there's a risk they will miss this place and it will remain not updated.

Comment (ShimShtein) [reply app/models/parameter.rb:86]: @ares, I think the problem is a bit different. I have tried to write it in the [comment](https://github.com/theforeman/foreman/pull/9966#issuecomment-1873305248) on this PR. 
TL;DR: we don't know the parameter type in the search function. I think we should solve it from there, it's not a problem with value serialization.

Comment (ares) [reply app/models/parameter.rb:86]: I agree, this solution would help only to some degree. If we'd go with this solution, I'd like to see the serializing logic on one place and instead of staying on the safe side, we should go case by case and see what the impact would be. But if we're not going this route and the resulting patch will be different, of course this is no longer relevant. The main principle remains valid (don't split the code just due to uncertainty of impact)

**Comment by girijaasoni** [comment]
Converting it to draft as 6.15 bugs are higher priority right now
